### PR TITLE
compensate injector flow based on fuel pressure

### PR DESCRIPTION
**Pull request by mck1117**
Automagically adjust injector flow based on sensed fuel rail pressure.

- enables fueling control on GDI
- enables proper fueling on fixed-rail-pressure fuel systems (particularly the NB Miata!)

### CODE REVIEW

File: firmware/controllers/algo/fuel/injector_model.cpp [hunk 1]
```diff
@@ -15,10 +17,51 @@ constexpr float convertToGramsPerSecond(float ccPerMinute) {
...
+		tsOutputChannels.debugFloatField2 = pressureRatio;
+		tsOutputChannels.debugFloatField3 = flowRatio;
+	}
+#endif // EFI_TUNER_STUDIO
+
+	return flowRatio;
+}
+
 float InjectorModel::getInjectorMassFlowRate() const {
 	// TODO: injector flow dependent upon rail pressure (and temperature/ethanol content?)
```
Comment (andreika-git) [inline firmware/controllers/algo/fuel/injector_model.cpp:59]: Maybe remove "rail pressure" part from this comment?

File: firmware/controllers/system/timer/single_timer_executor.cpp [hunk 2]
```diff
@@ -127,6 +127,13 @@ void SingleTimerExecutor::executeAllPendingActions() {
 	do {
 		efitick_t nowNt = getTimeNowNt();
 		didExecute = queue.executeOne(nowNt);
+
+		// If locked (ie, interrupts disabled), briefly enable/disable them
+		// to flush any pending interrupts
+		// This is safe to do here because it's the same as if the just executed
```
Comment (rusefillc) [inline firmware/controllers/system/timer/single_timer_executor.cpp:133]: does this great change belong in "compensate injector flow based on fuel pressure" PR? let's please separate?

File: unit_tests/tests/ignition_injection/test_injector_model.cpp [hunk 3]
```diff
@@ -53,3 +55,106 @@ TEST(InjectorModel, Deadtime) {
 	engine->sensors.vBatt = 7;
 	EXPECT_EQ(dut.getDeadtime(), 14);
 }
+
+struct TesterGetFlowRate : public InjectorModel {
```
Comment (rusefillc) [inline unit_tests/tests/ignition_injection/test_injector_model.cpp:59]: thank you for new unit tests!

File: firmware/controllers/system/timer/single_timer_executor.cpp (hunk 2 above)
Comment (mck1117) [reply firmware/controllers/system/timer/single_timer_executor.cpp:133]: oops, that wasn't supposed to sneak in

**Comment by rusefillc** [comment]
is that timer/scheduling magic intended or useful in master eventually or is that just test code?

**Comment by mck1117** [comment]
#1934 created for scheduler change, needs a bit more testing though (hence draft)

**Comment by mck1117** [comment]
Now tested on real hardware - the only real problem was that my fuel pressure sensor seems to indicate wrong :cry: (hardware problem, not rusefi software!)

**Comment by mck1117** [comment]
This probably closes #954 too!

**Pull request merged**